### ISSUE TITLE
Inconsistency when writing to StorageDict

### ISSUE THREAD

**Issue by adrianespejo** [bsc-dd/hecuba#195]
```python
In [4]: class URCA(StorageDict):
   ...:     '''
   ...:     @TypeSpec <<int>,int>
   ...:     '''
   ...:     

In [5]: a=URCA('hola.ciao')

In [6]: for i in range(10):
   ...:     for j in range (10):
   ...:         a[i]=j
   ...:         

In [7]: a
Out[7]: {0: 8, 1: 6, 2: 8, 3: 8, 4: 8, 5: 9, 6: 9, 7: 9, 8: 9, 9: 9}

a should be {0: 9, 1: 9, 2: 9, 3: 9, 4: 9, 5: 9, 6: 9, 7: 9, 8: 9, 9: 9}
```

**Comment by polsm91** [comment]
@cugni 
This is related to overriding the same value (a[i]).
We use an asynchronous, parallel, writing strategy without timestamps. Therefore, we can't enforce "last write wins" and we should change the current implementation.

Actually, we need to change the write strategy to solve #122

**Comment by cugni** [comment]
relates to #234 and #233

**Comment by polsm91** [comment]
Fixed by #233
